scripts/upsert/courses.ts

```diff
@@ -0,0 +1,36 @@
+import { Term } from '../../lib/term';
+import { differenceInDays, differenceInMinutes } from 'date-fns';
+import { upsertCourses } from '../../lib/courses';
+import { findLatestTask, createTask } from '../../lib/task';
+
+export const upsertCoursesScript = async (term: string, registrationDay: Date, dropDate: Date) => {
+  const today = new Date();
+
+  const daysUntilRegistration = differenceInDays(registrationDay, today);
+
+  const lastUpdated = await findLatestTask(`upsertCourses-${term}`);
+
+  const minutesSinceLastUpdate = differenceInMinutes(today, lastUpdated?.startedAt ?? -1);
+
+  // If over a week away from registration day, run task less frequently
+  if (daysUntilRegistration > 7) {
+    // If before drop date and courses have been updated within the last hour, return
+    if (today < dropDate && minutesSinceLastUpdate <= 60) return;
+    // If after drop date and courses have been updated within the last 24 hours, return
+    if (today > dropDate && minutesSinceLastUpdate <= 1440) return;
+  }
```

Comment:
Is another case not needed to stop updating after the drop date?

Reply:
Sure - I was just under the impressions we always want updated data, just less frequent! What's a good business logic for this case then?
Within a week of registration -> every 30min
Before drop date -> every hour
After drop date -> stop?